### DISCUSSION TITLE
Jest Testing with async errors

### DISCUSSION THREAD

**Discussion by nathanthibert** [vuejs/pinia#904] [Help and Questions] [answered]
I have a test where I'm rejecting a service call. We can use the documentation example. 

```js
 async registerUser(login, password) {
      try {
        this.userData = await api.post({ login, password })
        showTooltip(`Welcome back ${this.userData.name}!`)
      } catch (error) {
        showTooltip(error)
        // let the form component display the error
        return error
      }
    },
```

How do we test the return value of the error? I assume this hooks into the `$onAction(({onError})=>{})`. How do we test this in our store and or in our component?

**Comment by posva** [comment, marked as answer]
You can instantiate a regular store in a test and call the action and check the returned value:

```ts
expect(await store.registerUser(...)).toEqual(...)
```

**Reply by nathanthibert** [reply]
I love how easy this framework is. Amazing!
